client/gutenberg/extensions/publicize/store/reducer.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * Default Publicize reducer state
+ */
+const DEFAULT_STATE = {
+	connections: [],
+	services: null,
+};
+
+/**
+ * Reducer managing Publicize extension state.
+ *
+ * @param {Object} state  Current state.
+ * @param {Object} action Dispatched action.
+ *
+ * @return {Object} Updated state.
+ */
+const reducer = ( state = DEFAULT_STATE, action ) => {
+	switch ( action.type ) {
+		case 'SET_CONNECTIONS':
+			return {
+				...state,
+				connections: {
+					...state.connections,
+					[ action.postId ]: action.connections,
+				},
+			};
+		case 'REFRESH_CONNECTIONS':
+			return {
+				...state,
+				connections: DEFAULT_STATE.connections,
+			};
+		case 'SET_SERVICES':
+			return {
+				...state,
+				services: action.services,
+			};
+	}
+
+	return state;
+}
```

Comment:
We could compose this from two subreducers, `connections`, and `services`. Maybe material for a follow-up PR.

Reply:
Even better - addressed in ff6ac7ba58b07ac7906e5d7683577182a45af3b0.